util/validate/validation/index.mjs

```diff
@@ -19,10 +19,85 @@ import { localizeDependencies } from '../../shared/json-schema.mjs'
 import crocks from 'crocks'
 const { getPathOr } = crocks
 
+const addPrettyPath = (error, json) => {
+  const path = []
+  const root = json.title || json.info.title
+
+  let pointer = json
```

Comment:
Why?

Reply:
The code doesn't mutate the `json` variable or any of it's sub-object, either through the `json` reference nor the `pointer` reference.

The code is using `pointer` to walk through the JSON tree w/out loosing the original reference to the top of the tree.

You could argue that `pointer` isn't needed for this, but it allows for each code editing, for example if we wanted to add `json.info.title` to the returned string, we'd still have access to that part of the object. Seems iffy to reuse a passed in parameter variable to do something like this.

Reply:
Not quite sure what you mean about not losing the original reference to the top of the tree.

You are mutating `pointer`, and `json` (by reference) [right here](https://github.com/rdkcentral/firebolt-openrpc/blob/b37fdf1370ad481cae5c67c1b5490ab83d64a073/util/validate/validation/index.mjs#L30).

Reply:
nope, we're just changing what pointer points to. For example:


```javascript
> let json = {foo: {bar: { str: 'hello'}}}
undefined
> json
{ foo: { bar: { str: 'hello' } } }
> let pointer = json
undefined
> pointer
{ foo: { bar: { str: 'hello' } } }
> let pointer = json['foo']
Uncaught SyntaxError: Identifier 'pointer' has already been declared
> pointer = json['foo']
{ bar: { str: 'hello' } }
> pointer
{ bar: { str: 'hello' } }
> pointer = pointer['bar']
{ str: 'hello' }
> pointer
{ str: 'hello' }
> pointer = pointer['str']
'hello'
> pointer
'hello'
> json
{ foo: { bar: { str: 'hello' } } }
> 
```

^^ from node